plugins/module_utils/mso.py

```diff
@@ -853,6 +855,13 @@ def lookup_roles(self, roles, ignore_not_found_error=False):
             ids.append(dict(roleId=r.get("id"), accessType=access_type))
         return ids
 
+    def lookup_site_type(self, s):
+        """Get site type(AWS, AZURE or physical)"""
+        site_type = s.get("platform")
+        if site_type == "cloud":
+            self.cloud_provider_type = s.get("cloudProviders")[0]
```

Comment:
Is this valid for all the cloud providers? (Will indexing on 0 give the right element for all the providers)

Reply:
Yes, It is the same for all types of cloud providers.